dotnet/src/Connectors/Connectors.Onnx/OnnxRuntimeGenAIChatCompletionService.cs

```diff
@@ -0,0 +1,139 @@
+    {
+        var result = new StringBuilder();
+
+        await foreach (var content in this.RunInferenceAsync(chatHistory, executionSettings, cancellationToken).ConfigureAwait(false))
+        {
+            result.Append(content);
+        }
+
+        return new List<ChatMessageContent>
+        {
+            new(
+                role: AuthorRole.Assistant,
+                content: result.ToString())
+        };
+    }
+
+    /// <inheritdoc />
+    public async IAsyncEnumerable<StreamingChatMessageContent> GetStreamingChatMessageContentsAsync(ChatHistory chatHistory, PromptExecutionSettings? executionSettings = null, Kernel? kernel = null, CancellationToken cancellationToken = default)
+    {
+        await foreach (var content in this.RunInferenceAsync(chatHistory, executionSettings, cancellationToken).ConfigureAwait(false))
+        {
+            yield return new StreamingChatMessageContent(AuthorRole.Assistant, content);
+        }
+    }
+
+    private async IAsyncEnumerable<string> RunInferenceAsync(ChatHistory chatHistory, PromptExecutionSettings? executionSettings, CancellationToken cancellationToken)
+    {
+        OnnxRuntimeGenAIPromptExecutionSettings onnxRuntimeGenAIPromptExecutionSettings = OnnxRuntimeGenAIPromptExecutionSettings.FromExecutionSettings(executionSettings);
+
+        var prompt = this.GetPrompt(chatHistory, onnxRuntimeGenAIPromptExecutionSettings);
+        var tokens = this._tokenizer.Encode(prompt);
+
+        var generatorParams = new GeneratorParams(this._model);
+        this.ApplyPromptExecutionSettings(generatorParams, onnxRuntimeGenAIPromptExecutionSettings);
+        generatorParams.SetInputSequences(tokens);
+
+        var generator = new Generator(this._model, generatorParams);
+
+        while (!generator.IsDone())
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            yield return await Task.Run(() =>
+            {
+                generator.ComputeLogits();
+                generator.GenerateNextToken();
+
+                var outputTokens = generator.GetSequence(0);
+                var newToken = outputTokens.Slice(outputTokens.Length - 1, 1);
+                var output = this._tokenizer.Decode(newToken);
+                return output;
+            }, cancellationToken).ConfigureAwait(false);
+        }
+    }
+
+    private string GetPrompt(ChatHistory chatHistory, OnnxRuntimeGenAIPromptExecutionSettings onnxRuntimeGenAIPromptExecutionSettings)
+    {
+        var promptBuilder = new StringBuilder();
+        foreach (var message in chatHistory)
+        {
+            promptBuilder.Append($"<|{message.Role}|>\n{message.Content}");
+        }
+        promptBuilder.Append("<|end|>\n<|assistant|>");
+
+        return promptBuilder.ToString();
+    }
+
+    private void ApplyPromptExecutionSettings(GeneratorParams generatorParams, OnnxRuntimeGenAIPromptExecutionSettings onnxRuntimeGenAIPromptExecutionSettings)
```

Comment:
This to give more clarity. 

```suggestion
    private void UpdateGeneratorParamsFromPromptExecutionSettings(GeneratorParams generatorParams, OnnxRuntimeGenAIPromptExecutionSettings onnxRuntimeGenAIPromptExecutionSettings)
```